packages/runtime/src/components/builtin/Slot/SlotRoot.tsx

```diff
@@ -0,0 +1,21 @@
+'use client'
+
+import { ReactNode, useContext } from "react"
+import { PropsContext } from "../../../next/components/MakeswiftSlot"
+
+type Props = {
+  children: ReactNode
+  showFallback: boolean
+}
+
+export const SlotRoot = (props: Props) => {
+  const ctx = useContext(PropsContext)
+
+  return (
+      <>
+        { props.showFallback ? props.children : ctx.fallback}
```

Comment:
Out of curiosity, why does `fallback` come from a context?

Reply:
The situation that prompted this is that the new slot component is used in hardcoded routes, but requires interactivity in the builder. The implication of this is that its props need to come from two different sources - some props need to come from controls (i.e. the content that is dragged into the slot), and other props need to be hardcoded (i.e. the snapshot ID). So the props that are hardcoded are being passed to SlotRoot via context, and the other props are coming from the controllers as usual.